scripts/doppler_doctor.py

```
@@ -0,0 +1,282 @@
+        for m in _COMPOSE_ENV_FILE_RE.finditer(text):
+            line_no = text[: m.start()].count("\n") + 1
+            violations.append(
+                f"{rel}:{line_no}: forbidden `env_file:` "
+                f"directive — wrap launch in `doppler run --` instead"
+            )
+    return violations
+
+
+def _git_tracked_paths(root: Path) -> list[str]:
+    """Repo-relative paths from ``git ls-files`` (skips untracked local ``.env``)."""
+    proc = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)